spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamDeploymentController.java

```diff
@@ -274,10 +274,20 @@ else if (hasOutput) {
 		for (Map.Entry<String, String> entry : streamDeploymentProperties.entrySet()) {
 			if (entry.getKey().startsWith(modulePrefix)) {
 				if (entry.getKey().startsWith(producerPropertyPrefix)) {
+					// Set per-binding properties
+					moduleDeploymentProperties.put(BindingProperties.OUTPUT_BINDING_KEY_PREFIX +
+							entry.getKey().substring(producerPropertyPrefix.length()), entry.getValue());
+					// Set ChannelBindingServiceProperties' producer properties
+					// todo: Once the producer/consumer properties are moved to per-binding properties, following step
```

Comment:
In other words, while "consumer" and "producer" might make sense within the scope of the dataflow DSL, we need to map those to the corresponding bindings without carrying over the terms "producer" or "consumer" on the `spring-cloud-stream` side (where they would be superfluous). On the dataflow side, we do know the channel names (input/output) for which are providing the properties. It seems that we need to map from `producer.*` to the output channel binding properties and map from `consumer.*` to the input channel binding properties. The key following `prodcuer/consumer.` should match that expected in `BindingProperties` itself.